From: Michael Olbrich <m.olbrich@pengutronix.de>
Date: Sat, 2 Mar 2024 16:44:19 +0100
Subject: [PATCH] gst-python: fix cross-compiling

When cross-compiling, looking in the root filesystem of the build host to find
the Python library makes no sense. Instead, assume that the directory and file
name provided by Python are correct and use those without checking.
---
 meson.build | 24 ++++++++++++++----------
 1 file changed, 14 insertions(+), 10 deletions(-)

diff --git a/meson.build b/meson.build
index f4dfcf34ae11..539845f4d3fb 100644
--- a/meson.build
+++ b/meson.build
@@ -87,17 +87,21 @@ else
   endif
 endif
 pylib_fname = ''
-foreach loc: pylib_locs
-  foreach fname: pylib_fnames
-    fpath = loc / fname
-    debug(f'Looking for Python library at: @fpath@')
-    if fsmod.exists(fpath)
-      pylib_fname = fname
-      message(f'PY_LIB_FNAME = @fname@ (@loc@)')
-      break
-    endif
+if meson.is_cross_build()
+  pylib_fname = pylib_locs[0] / pylib_fnames[0]
+else
+  foreach loc: pylib_locs
+    foreach fname: pylib_fnames
+      fpath = loc / fname
+      debug(f'Looking for Python library at: @fpath@')
+      if fsmod.exists(fpath)
+        pylib_fname = fname
+        message(f'PY_LIB_FNAME = @fname@ (@loc@)')
+        break
+      endif
+    endforeach
   endforeach
-endforeach
+endif
 if pylib_fname == ''
   message('Could not find python library to load, will try loading at runtime')
 endif
